pkg/apis/k0s/v1beta1/api.go

```diff
@@ -35,6 +35,9 @@ type APISpec struct {
 	// Local address on which to bind an API
 	Address string `json:"address"`
 
+	// Local address on which to bind the kube-apiserver. Defaults to 0.0.0.0.
+	BindAddress string `json:"bindAddress,omitempty"`
```

Comment:
We might want to add some kubebuilder validation here to reject nonsensical values already at the API server level. Sure, a regex that's correctly checking for IPv4/IPv6 compliance will probably be one megabyte in size or so, but something short that rejects obviously wrong stuff might be helpful nevertheless?

Reply:
what are you thinking of? something like `^([0-9.]+|[0-9a-f:]+)$` would at least ensure that the correct alphabet is being used while keeping it simple.

Reply:
Yeah, something like this. Although I have a gut feeling that the current regex will break IPv6 link-local addresses. Maybe my idea wasn't the best here :smile:. A pity that there's no "ip" format in the OpenAPI validator, just "ipv4" _or_ "ipv6" (at least as far as I reckon). A workaround would be to have `bindAddress: {type: IPv4, ipv4: 127.0.0.1}` and `bindAddress: {type: IPv6, ipv6: "::1"}`, just to have two distinct fields to put the format tags on. That'd be a _very_ artificial distinction, though.

Overall, we might be better off without server-side validation, unfortunately, just to not introduce unnecessary bugs.

Reply:
> Although I have a gut feeling that the current regex will break IPv6 link-local addresses.

that's why I added the `%` in a later revision, but I'm still not familiar enough with the related RFCs to be certain it covers everything.

I also lean towards not risking it.

Reply:
Is this something that could be deferred to after this PR ?

Reply:
Yeah, let's just skip validation right now, and just :crossed_fingers: